src/recommender.py

```diff
@@ -194,11 +195,20 @@ def _send_action(self, speaker_id, action):
       # sending action to phone
       try:
         # send prequestion 22
-        url = phone_url + '/?q={%22id%22:%22' + str(speaker_id) + '%22,%22c%22:%22startsurvey%22,%22suid%22:%22' + '22' + '%22,%22server%22:%22' + \
-          server_url + '%22,%22androidid%22:%22' + androidid + '%22,%22empathid%22:%22' + \
-          pre_empathid + '%22,%22alarm%22:%22' + alarm + '%22}'
-        # webbrowser.open(url)  # to open on browser
 
+        url_dict = {
+          "id": str(speaker_id),
+          "c":"startsurvey",
+          "suid": '22',
+          "server": server_url,
+          "androidid": androidid,
+          "empathid": pre_empathid,
+          "alarm": alarm
+        }
+
+        url_string = json.dumps(url_dict)
+        url =phone_url + '/?q='+ url_string
+        url = urllib.parse.quote(url,safe=':?={}/') #encoding url quotes become %22
```

Comment:
actually, we only want to escape the value of the query argument `q`
A suggestion is to `urllib.parse.quote(json.dumps(url_dict), safe=':?={}/')` at first, so we don't need to escape `phone_url` and the string `'/?q='` (u may find you can have less `safe` character, say `?`)

then append the encoded `q_dict_string` like before to form the `url`